### ISSUE TITLE
Fetching access token fails

### ISSUE THREAD

**Issue by hoehlenwolf** [ihabunek/twitch-dl#14]
Hey,
first of all this is a great tool. Fast, simple and does what it is supposed to. :)

However recently I've stumbled upon the issue, that downloading a specific VOD fails shortly after trying to fetch the access token.
For example:
```console
twitch-dl.exe download 510977664 --no-color
Looking up video...
Found: Riot Report Episode 4 by Riot Games
Fetching access token...
Traceback (most recent call last):
  File "g:\programme\python\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "g:\programme\python\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "G:\tools\twitch-dl.exe\__main__.py", line 9, in <module>
  File "g:\programme\python\lib\site-packages\twitchdl\console.py", line 148, in main
    args.func(**args.__dict__)
  File "g:\programme\python\lib\site-packages\twitchdl\commands.py", line 230, in download
    access_token = twitch.get_access_token(video_id)
  File "g:\programme\python\lib\site-packages\twitchdl\twitch.py", line 71, in get_access_token
    return authenticated_get(url).json()
  File "g:\programme\python\lib\site-packages\twitchdl\twitch.py", line 19, in authenticated_get
    response.raise_for_status()
  File "g:\programme\python\lib\site-packages\requests\models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 410 Client Error: Gone for url: https://api.twitch.tv/api/vods/510977664/access_token
```

This happens to all videos I've tried so far.
Any ideas what causes this?

Thanks in advance =)

**Comment by Seraphyx** [comment]
I am getting this error too. Has the Twitch API changed perhaps?

**Comment by ihabunek** [comment]
Yeah, they keep changing the API. It's a pain having to deal with a moving target. I'll look into it when I get some free time.

The v5 API has been deprecated for ages, but has the required functionality. The current API is called "The New Twitch API" (terrible name, will the next one be "new new FINAL API?) and last time I looked it was not possible to implement twitch-dl using that. And I'm also using an undocumented endpoint. :)

I don't know if twitch-dl is worth the hassle if they keep moving things around all the time. But let's try it for a bit longer.

**Comment by hoehlenwolf** [comment]
I did look into that a bit myself, seems like they don't want people to download VODs for later offline-watching purposes, because they can't make you watch ads then :D
I also did stumble across the hidden, undocumented famous singer's endpoint. ;)
It seems to not work anymore, as far as I've tried.

Maybe, as a (temporary) alternative, the project [youtube-dl](https://github.com/ytdl-org/youtube-dl) might work. It manages to download videos from twitch (and a lot of other sources), but it is extremely slow and doesn't utilize the full bandwidth available. Your's was far better in my opinion :)

Maybe it might help to come up with a workaround here though, that would be amazing

Thank you for your time and effort :)

**Comment by ihabunek** [comment]
TL;DR: This is fixed in v.1.5.0.

So I finally took the time to sit down and see how `youtube-dl` and others handle this issue. Turns out they do the exact same thing that `twitch-dl` does.

The only difference is the Twitch [Client ID](https://dev.twitch.tv/docs/v5#getting-a-client-id) they use. I created one for `twitch-dl` some years ago, and with it I can no longer get the access token. The others are using the Client ID which twitch.tv uses itself and using that one they can still get the access token. They're [not the only one](https://github.com/search?p=2&q=kimne78kx3ncx6brgo4mv6wki5h1ko&type=Code).

I have no expectation of this fix working in the long run, they're bound to break something else, but enjoy while it works. \o/
